SW.SecurityService.Core/Services/AuthenticationService.cs

```diff
@@ -1,26 +1,65 @@
 ﻿namespace SW.SecurityService.Core.Services
 {
+    using System;
+    using Newtonsoft.Json;
+    using SW.SecurityService.Core.Models;
+    using SW.SecurityService.Core.Providers;
     using SW.SecurityService.CredentialRepository.Repository;
 
     public class AuthenticationService : IAuthenticationService
     {
+        private readonly ITokenService tokenService;
+        private readonly ITokenProvider tokenProvider;
+        private readonly IDateTimeProvider dateTimeProvider;
         private readonly ICredentialRepository credentialsRepository;
 
-        public AuthenticationService(ICredentialRepository credentialsRepository)
+        public AuthenticationService(
+            ITokenService tokenService,
+            ITokenProvider tokenProvider,
+            IDateTimeProvider dateTimeProvider,
+            ICredentialRepository credentialsRepository)
         {
+            this.tokenService = tokenService;
+            this.tokenProvider = tokenProvider;
+            this.dateTimeProvider = dateTimeProvider;
             this.credentialsRepository = credentialsRepository;
         }
 
-        public bool IsValidCredentials(string user, string password)
+        public AuthenticationAnswer AuthenticateUser(string userName, string password)
         {
-            var actualCredentials = this.credentialsRepository.GetCredential(user);
+            var actualCredentials = this.credentialsRepository
+                .GetCredential(userName);
+
+            var authenticationAnswer = new AuthenticationAnswer();
+            
+            if (actualCredentials is null)
+            {
+                authenticationAnswer.nonExistenLogin = true;
```

Comment:
Maybe we should create Enum that represent state of `AuthenticationAnswer` smth like 
```
enum AuthenticationAnswerStatus
{
    Authorized = 1,
    WrongPassword = 2,
    UserNotExist = 3,
    SystemFault = 4
}
```
And instead of multiple bool properties use this enum as status?